src/bmp.hpp

```diff
@@ -15,7 +15,9 @@
 #include <cstddef>
 #include <cstdint>
 #include <iostream>
+#include <optional>
 #include <vector>
+#include "immintrin.h"
```

Comment:
do you actually need this here, or only in `bmp.cpp`?  (and by convention, `#include <...>` style is used for all kinds of include files that come "from the implementation", not just for the standard headers)